src/hooks/useGreetingMessage.js

```diff
@@ -0,0 +1,49 @@
+import React, {useEffect, useState, useContext} from "react"
+import {AppContext} from "../context/AppContext";
+import {VoiceCommandsContext} from "../context/VoiceCommandsContext";
+import {LoggingContext} from "../context/LoggingContext";
+
+export const useGreetingMessage = () => {
+
+    const voiceContext = useContext(VoiceCommandsContext);
+    const loggingContext = useContext(LoggingContext).logger;
+
+    const [style, setStyle] = useState(0);
+    const [name, setName] = useState("");
+
+    const greetingStyles = [
+        // 0 Standard Greeting
+        `Hello ${name}, welcome back!`,
```

Comment:
maybe a null check for the name. As I understand you're setting name outside of the hook